unitstatushandler/softwaremanager.go

```diff
@@ -259,8 +259,7 @@ func (manager *softwareManager) processDesiredServices(
 downloadServiceLoop:
 	for _, desiredService := range desiredServices {
 		for _, service := range allServices {
-			if desiredService.ID == service.ID && desiredService.AosVersion == service.AosVersion &&
-				service.Status == cloudprotocol.InstalledStatus {
+			if desiredService.ID == service.ID && desiredService.AosVersion == service.AosVersion {
```

Comment:
Unclear at which circumstance it may happen if we do not process new desired status while current one is still installing.  Let's go through this case again.

Reply:
instead of checking the service installed status use the error check